maroon/src/app.rs

```diff
@@ -1,22 +1,55 @@
 use crate::p2p_interface::{Inbox, NodeState, Outbox, P2PChannels};
 use common::{
+    async_interface::{AsyncInterface, ReqResPair},
     gm_request_response::Transaction,
     range_key::{self, KeyOffset, KeyRange, TransactionID},
 };
 use libp2p::PeerId;
-use log::{debug, error, info, warn};
+use log::{error, info};
 use std::{
     collections::{HashMap, HashSet},
     num::NonZeroUsize,
     time::Duration,
 };
 use tokio::sync::oneshot;
 
-pub struct App {
+pub enum Request {
+    GetState,
+}
+#[derive(Debug, PartialEq, Eq)]
+pub enum Response {
+    State(CurrentOffsets),
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct CurrentOffsets {
+    pub self_offsets: HashMap<KeyRange, KeyOffset>,
+    pub consensus_offset: HashMap<KeyRange, KeyOffset>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Params {
+    /// how often node will send state info to other nodes
+    pub advertise_period: std::time::Duration,
     /// minimum amount of nodes that should have the same transactions(+ current one) in order to confirm them
-    consensus_nodes: NonZeroUsize,
+    pub consensus_nodes: NonZeroUsize,
```

Comment:
Right, so we have different sets of params, and I'd say separating them into "algorithmic/logical" and "network/practical" would be nice. Why not a separate `pub struct ConsensusAlgoParams` in a separate `lib/consensus` crate with its own test suite?